### PR TITLE
categories

### PR DESCRIPTION
**Pull request by samson-favour**
add changes to the category service

### CODE REVIEW

File: src/category/category.controller.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { Category } from 'src/entities/category.entity';
+import { CategoryService } from './category.service';
+import { CreateCategoryDto } from './dto/create.category';
+
+@Controller('categories')
+export class CategoryController {
+  constructor(private readonly categoryService: CategoryService) {}
+
+  @Get('/getcategories')
+  @ApiOperation({ summary: 'Get an array of al categories' })
```
Comment (Chimexx) [inline src/category/category.controller.ts:13]: Fix this typo

File: src/category/category.controller.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    return {
+      categories: categories.map((category) => category.name),
+    };
+  }
+  @Post('category')
+  @ApiOperation({ summary: 'create a category' })
+  async CreateCategory(
+    @Body() categoryDto: CreateCategoryDto,
+  ): Promise<Category> {
+    return await this.categoryService.createCategory(categoryDto);
```
Comment (Chimexx) [inline src/category/category.controller.ts:25]: Use zapi response object

File: src/category/category.controller.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  @Get('/:id')
+  @ApiOperation({ summary: 'Get a single categories with a related apis' })
+  getCategoryById(@Param('id') id: string) {
+    return this.categoryService.getCategoryById(id);
```
Comment (Chimexx) [inline src/category/category.controller.ts:31]: Please Identify everywhere you need to use zapi response object

File: src/category/category.service.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+export class CategoryService {
+  constructor(
+    @InjectRepository(CategoryRepository)
+    private categoryRepository: CategoryRepository,
+    @InjectRepository(APIRepository)
+    private apiRepository: APIRepository,
+  ) {}
+
+  async getAllCategory(): Promise<Category[]> {
+    const categories = await this.categoryRepository.find({});
```
Comment (Chimexx) [inline src/category/category.service.ts:19]: What is the curly braces doing here?

File: src/category/category.service.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  async createCategory(categorydto: CreateCategoryDto): Promise<Category> {
+    return await this.categoryRepository.createCategory(categorydto);
+  }
+
+  async getCategoryById(id: string) {
+    try {
+      const found = this.categoryRepository.findOne({ id });
+      return found;
+    } catch (error) {
+      if (error.code === '22P02') {
```
Comment (Chimexx) [inline src/category/category.service.ts:31]: What if the error code doesn't match 22P02?

File: src/category/category.service.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        );
+      }
+    }
+  }
+
+  async deleteCategory(id: string): Promise<void> {
+    const result = await this.categoryRepository.delete({ id });
+    if (result.affected === 0) {
+      throw new NotFoundException(
+        ZapiResponse.BadRequest(`Task with  this ID - ${id} not found`),
```
Comment (Chimexx) [inline src/category/category.service.ts:43]: Please re write this logic, I expect you to take clues from existing codes in the repo, but you are not doing that. What exactly are you deleting here?

File: src/category/dto/create.category.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { IsNotEmpty, IsString } from 'class-validator';
+
+export class CreateCategoryDto {
+  @IsString()
+  @IsNotEmpty()
+  name: string;
+
```
Comment (Chimexx) [inline src/category/dto/create.category.ts:7]: use @ApiProperty() decorator

File: src/database/repository/category.repository.ts [hunk 8]
```diff
@@ -1,5 +1,23 @@
...
-export class CategoryRepository extends Repository<Category> {}
+export class CategoryRepository extends Repository<Category> {
+  async createCategory(categorydto: CreateCategoryDto): Promise<Category> {
+    try {
+      const { name, description } = categorydto;
+      const category = this.create({ name, description });
+      await this.save(category);
+      return category;
+    } catch (error) {
+      throw new BadRequestException(
```
Comment (Chimexx) [inline src/database/repository/category.repository.ts:16]: Why are you doing this inside the repository file?

**Review by Chimexx — CHANGES_REQUESTED** [review]

File: src/category/category.service.ts (hunk 6 above)
Comment (DorcasOJ) [reply src/category/category.service.ts:43]: I think you should follow the norms of creating repository file in the main branch

**Review by DorcasOJ — CHANGES_REQUESTED** [review]
Kindly fix the requested changes

**Pull request closed without merging**